artifact/v3/manifest/manifest.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_ARTIFACT_V3_MANIFEST_PARSER_HPP
+#define MENDER_ARTIFACT_V3_MANIFEST_PARSER_HPP
+
+#include <string>
+#include <unordered_map>
+
+#include <common/expected.hpp>
+#include <common/error.hpp>
+#include <common/io.hpp>
+
+namespace artifact {
+namespace v3 {
+namespace manifest {
+
+using namespace std;
+
+namespace expected = mender::common::expected;
+namespace io = mender::common::io;
+
+struct Manifest {
+	unordered_map<string, string> m;
```

Comment:
I wonder if we need a `vector` instead, because order matters when we start reading the payloads later.

Although if you can look up by name, then maybe it is enough to store the index together with the checksum, so that you can check what it is while reading.

Reply:
I'd rather keep it like this actually, we do lookups by name.